### PR TITLE
Tools implemented and update to simplify release process

### PR DESCRIPTION
**Pull request by Czaki**
# Description

In this PR I will keep all scripts used to automatize v0.4.18 release process. After it ends it should be revisited and decided if some of them should be preserved for future releases. 

# References

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

File: tools/release/generate_issue_content.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+LAST_TAG="v0.4.17"
+NEW_RELEASE="0.4.18"
+FUTURE_RELEASE="0.5.0"
```
Comment (Carreau) [inline tools/release/generate_issue_content.sh:5]: You might want to use `read` to ask the user to input value. 

Maybe https://github.com/ipython/ipython/blob/main/tools/release_helper.sh can be of inspiration.

File: .gitignore [hunk 2]
```diff
@@ -3,6 +3,8 @@ __pycache__/
 *.py[cod]
 *$py.class
 
+tools/release/issue.md
```
Comment (Carreau) [inline .gitignore:6]: Missing file ?

File: tools/release/filter_pr_that_may_be_selected.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+setup_cache()
+
+repository = get_repo()
+
+milestone = get_milestone(args.milestone)
+
+label = repository.get_label(args.label) if args.label else None
+
+if args.skip_triaged:
+    triage_label = repository.get_label("triaged-0.4.18")
```
Comment (Carreau) [inline tools/release/filter_pr_that_may_be_selected.py]: Hardcoded label.

**Review by Carreau — APPROVED** [review]
I don't think we need this to be perfect, but we can iterate for each release.

File: .gitignore (hunk 2 above)
Comment (Czaki) [reply .gitignore:6]: I do not understand. I put output of `generate_issue_content.sh` in this file.

Comment (Carreau) [reply .gitignore:6]: So you are doing `sh generate_issue_content.sh > tools/release/issue.md` ? But where ? It's not documented anywhere, or maybe I'm missing something ?

Comment (Czaki) [reply .gitignore:6]: it is not documented, as I do not add yet any readme with documentation.

Comment (Carreau) [reply .gitignore:6]: ok.

**Pull request closed without merging**